### ISSUE TITLE
Images from CMS

### ISSUE THREAD

**Issue by David-Slack** [MinJieLiu/react-photo-view#191]
I have a bunch of pages and each page has content from a CMS rendered with:

```
<article className="content" dangerouslySetInnerHTML={{ __html: content.body }}></article>
```

Can I use the PhotoViewer with these images and how would I do it?

Thanks

**Comment by xavier-fenton** [comment]
What's your current stack? Because I've been able to do it with Sanity Studio

**Comment by David-Slack** [comment]
I use Next.js and content comes in via a restful api from Drupal CMS. Content comes in server side Next.js, is rendered and passed down so should be available in the DOM or virtual DOM, but because it's a cms we cannot wrap the images in any elements.

I've not looked at this issue in a long time. I ended up coding my own with css and js. It simply finds any images and changes the elements with captions and click for full size. Means I bring in a larger image than needed and reduce the size with css until clicked on

**Comment by xavier-fenton** [comment]
Oh, I think I understand, is it not possible because the content from the CMS is fully customisable? Maybe in the way that the user of the CMS has full control of the way it looks etc?

**Comment by David-Slack** [comment]
This is correct. The body of the content has a WYSIWYG with an image uploader. There are main and sub images, and these work fine with the viewer, but the body images don't work e.g.
https://www.huytonweb.com/projects/huyton-web-services-bespoke-website

The main and sub images at the top work fine, but scroll down and click on the other images and you'll see I've had to add my own viewer.

I could probably wrap the content in the correct elements on the server in some way, but I was hoping there was another way to do this
